_pages/dev/ssr/server-side-rendering-error-handling.md

```diff
@@ -0,0 +1,422 @@
+}
+```
+
+If you find the functionality of `defaultExpressErrorHandlers` sufficient but want a customize how one type of error is handled, you can provide a custom error handler before the `defaultExpressErrorHandlers`. Make sure to call `next(err)` in the custom error handler to let the `defaultExpressErrorHandlers` take care of the error not handled by the custom handler.
+The following is an example of how to provide a custom error handler for `CmsPageNotFoundOutboundHttpError`:
+```ts
+//custom-error-handler.ts
+export const customCmsPageNotFoundErrorHandler: ErrorRequestHandler = async (
+  err,
+  _req,
+  res,
+  next
+) => {
+  const errorPage = /*...*/;
+  if (err instanceof CmsPageNotFoundOutboundHttpError) {
+    res.status(HttpResponseStatus.NOT_FOUND).send(errorPage);
+  } else {
+    next(err);
+  }
+};
+```
+
+```ts
+//server.ts
+import { APP_BASE_HREF } from '@angular/common';
+import {
+  ngExpressEngine as engine,
+} from '@spartacus/setup/ssr';
+import { customErrorPageErrorHandlers } from './custom-error-handler';
+
+import express from 'express';
+import { existsSync, readFileSync } from 'node:fs';
+import { join } from 'path';
+
+export function app(): express.Express {
+  const server = express();
+  const distFolder = join(process.cwd(), 'dist/storefrontapp');
+  const indexHtml = existsSync(join(distFolder, 'index.original.html'))
+    ? join(distFolder, 'index.original.html')
+    : join(distFolder, 'index.html');
+  const indexHtmlContent = readFileSync(indexHtml, 'utf-8');
+
+  /* ... */
+
+  server.use(customErrorPageErrorHandlers); // <-- custom error handler for 404 provided before default error handlers.
+  server.use(defaultExpressErrorHandlers(indexHtmlContent));
+
+  return server;
+}
+```
+
+## Strict error handling for HTTP and NgRx
+
+ To seal the Angular application from the asynchronous errors that occur in NgRx flow and HTTP calls during the rendering process, Spartacus provides set of tools to handle them.
+
+### Handling HTTP errors
+
+To handle HTTP errors, Spartacus provides `HttpErrorHandlerInterceptor` that catches the HTTP errors and forward them to the `CxErrorHandler`. The interceptor by default helps to distinguish two types of error by wrapping them with custom error classes before passing them to the `CxErrorHandler`:
+- `OutboundHttpError` - represents an outbound HTTP error that occurs when communicating with the backend.
+- `CmsPageNotFoundOutboundHttpError` - represents an outbound HTTP error specific to a CMS page not found. Extends the base OutboundHttpError class.
+
+Thanks to this, `defaultExpressErrorHandlers` middleware can handle the errors properly and return the correct status code to the client.
+
+**Note:** Make sure your custom interceptors that caches error are provided after `HttpErrorHandlerInterceptor` (so they run as next in the sequence) or they rethrow the error downstream to next interceptors. Otherwise, `HttpErrorHandlerInterceptor` will not be noticed about the error.
```

Comment:
typo? `caches` vs `catches` ?
```suggestion
**Note:** Make sure your custom Angular HTTP interceptors are provided after `HttpErrorHandlerInterceptor` (so they run as next in the sequence). If they catch any error, make sureto rethrow the error downstream to next interceptors. Otherwise, Spartacus `HttpErrorHandlerInterceptor` will not capture such an error, and therefore it won't be able to propagate it to ExpressJS layer and cause sending and appropriate HTTP error status code to the client of the SSR.  
```